docs/pages/dictionary.mdx

```diff
@@ -0,0 +1,262 @@
+
+### Value-type (`string`, `uint`, etc.)
+
+[Value-type](https://docs.soliditylang.org/en/latest/types.html#value-types) state variables such as `bool`, `uint256`, and `string` are represented by a singleton table, a table with no key that only contains a single record.
+
+<details>
+
+<summary>Code example</summary>
+
+```solidity filename="Solidity" {2}
+contract Example {
+  string name;
+}
+```
+
+```typescript filename="MUD" {6-11}
+import { defineWorld } from "@latticexyz/world";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Name: {
+      schema: {
+        name: "string",
+      },
+      key: [],
+    },
+  },
+});
+```
+
+</details>
+
+### Structures
+
+A [`struct`](https://docs.soliditylang.org/en/latest/types.html#structs) state variable wraps multiple variables together. You can create a singleton table with multiple value fields.
+
+<details>
+
+<summary>Code example</summary>
+
+```solidity filename="Solidity" {2-7}
+contract Example {
+  struct Person {
+    string name;
+    address personalWallet;
+    uint balance;
+  }
+  Person deployer;
+}
+```
+
+```typescript filename="MUD" {6-13}
+import { defineWorld } from "@latticexyz/world";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Deployer: {
+      schema: {
+        personalWallet: "address",
+        balance: "uint256",
+        name: "string",
+      },
+      key: [],
+    },
+  },
+});
+```
+
+</details>
+
+### Mappings
+
+A [mapping type](https://docs.soliditylang.org/en/latest/types.html#mapping-types) maps between a key type and a value type, which is pretty much the same as what MUD tables do.
+
+<details>
+
+<summary>Single key mapping</summary>
+
+```solidity filename="Solidity" {2}
+contract Example {
+  mapping(address => uint) public balances;
+}
+```
+
+```typescript filename="MUD" {6-12}
+import { defineWorld } from "@latticexyz/world";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Balances: {
+      schema: {
+        owner: "address",
+        balance: "uint256",
+      },
+      key: ["owner"],
+    },
+  },
+});
+```
+
+</details>
+
+<details>
+
+<summary>Multiple key mapping</summary>
+
+```solidity filename="Solidity" {2}
+contract Example {
+  mapping(address => mapping(address => uint256)) public allowances;
+}
+```
+
+```typescript filename="MUD" {6-13}
+import { defineWorld } from "@latticexyz/world";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Allowances: {
+      schema: {
+        owner: "address",
+        spender: "address",
+        balance: "uint256",
+      },
+      key: ["owner", "spender"],
+    },
+  },
+});
+```
+
+</details>
+
+<Callout type="info" emoji="📝">
+  MUD key fields have to be fixed-length.
+  This means that some mappings, for example `mapping(string => uint) balances;`, cannot be directly translated to a MUD table.
+
+An easy workaround is to use the hash of the string as the key instead of the string itself.
```

Comment:
```suggestion
An easy workaround is to use the hash of the string as the key instead of the string itself (which is exactly what happens under the hood of vanilla Solidity with the keys of a mapping).
```